esmf_regrid/tests/unit/experimental/unstructured_scheme/test_MeshToGridESMFRegridder.py

```diff
@@ -59,6 +61,54 @@ def _add_metadata(cube):
     assert expected_cube == result
 
 
+def test_multidim_cubes():
+    """
+    Test for :func:`esmf_regrid.experimental.unstructured_scheme.MeshToGridESMFRegridder`.
+
+    Tests with multidimensional cubes. The source cube contains
+    coordinates on the dimensions before and after the mesh dimension.
+    """
+    mesh = _full_mesh()
+    mesh_length = mesh.connectivity(contains_face=True).shape[0]
+
+    h = 2
+    t = 3
+    height = DimCoord(np.arange(h), standard_name="height")
+    time = DimCoord(np.arange(t), standard_name="time")
+
+    mesh_cube = Cube(np.zeros([t, mesh_length, h]))
+    mesh_coord_x, mesh_coord_y = mesh.to_MeshCoords("face")
+    mesh_cube.add_aux_coord(mesh_coord_x, 1)
+    mesh_cube.add_aux_coord(mesh_coord_y, 1)
+    mesh_cube.add_dim_coord(time, 0)
+    mesh_cube.add_dim_coord(height, 2)
+
+    n_lons = 6
+    n_lats = 5
+    lon_bounds = (-180, 180)
+    lat_bounds = (-90, 90)
+    tgt = _grid_cube(n_lons, n_lats, lon_bounds, lat_bounds, circular=True)
+
+    src_cube = mesh_cube.copy()
+    src_cube.transpose([1, 0, 2])
+    regridder = MeshToGridESMFRegridder(src_cube, tgt)
+    result = regridder(mesh_cube)
+
+    # Lenient check for data.
+    expected_data = np.zeros([t, n_lats, n_lons, h])
+    assert np.allclose(expected_data, result.data)
```

Comment:
Would it be better to do these tests with ones rather than zeros?  Would we catch more errors?